north/src/manifest.rs

```diff
@@ -122,12 +122,80 @@ pub struct CGroups {
 }
 
 // TODO: Remove?
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Resource {
     pub name: Name,
     pub version: Version,
     pub dir: std::path::PathBuf,
-    pub mountpoint: std::path::PathBuf,
+}
+
+impl FromStr for Resource {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let re = regex::Regex::new(r"(?P<name>\w+):(?P<version>[\d.]+)(?P<dir>[\w/]+)?")
+            .context("Wrong regular expression")?;
+
+        let caps = re
+            .captures(&s)
+            .ok_or_else(|| anyhow!("Invalid resource: {}", s))?;
+
+        let name = caps
+            .name("name")
+            .map(|m| m.as_str().to_owned())
+            .unwrap_or_default();
```

Comment:
Can this happen? I think not because otherwise the regex wouldn't have matched but I'm not sure.